landlab/components/landslides/landslide_probability.py

```diff
@@ -658,3 +858,34 @@ def _calculate_HSD_recharge(self, i):
             Re_adj = Re_temp * fract_temp
             store_Re = np.vstack((store_Re, np.array(Re_adj)))
         self._Re = np.sum(store_Re, 0)
+
+    def _calculate_HSD_groundwater_depth(self, i):
+        """Method to calculate groundwater depth.
+
+        This method calculates the resultant groundwater depth at node i of the
+        model domain, using depth to water table. Output is a numpy array
+        of depth to groundwater at node i.
+        """
+        self._hw_dist = self._hs_mode - self._HSD_dict[i]
+        self._hw_dist[
+            np.where(self._hw_dist < 0)
+        ] = 0  # no water in soil column when De input > soil thickness
+
+        # interpolate
+        self._interpolate_HSD_array()
+        self._De = self._HSD_dict[i]
+
+    def _build_unitgrid(rowcol, edgelength):
```

Comment:
This looks like a function that should not be in the component to me. Happy to have a second opinion from @mcflugen